aspnetcore/grpc/retries.md

```diff
@@ -0,0 +1,137 @@
+| BackoffMultiplier | The backoff will be multiplied by this value after each retry attempt and will increase exponentially when the multiplier is greater than 1. A value is required and must be greater than 0. |
+| RetryableStatusCodes | A collection of status codes. A gRPC call that fails with a matching status will be automatically retried. For more information about status codes, see [Status codes and their use in gRPC](https://grpc.github.io/grpc/core/md_doc_statuscodes.html). At least one status code is required. |
+
+## Hedging
+
+Hedging is an alternative retry strategy. Hedging enables aggressively sending multiple copies of a single gRPC call without waiting for a response. Because hedged gRPC calls may be executed multiple times on the server, it is important that hedging is only enabled for methods that are safe to execute multiple times without adverse affect.
+
+Hedging has pros and cons when compared to retries. An advantage to hedging is it might return a successful result faster. It allows for multiple simultaneously gRPC calls and will complete when the first successful result is available. A disadvantage to hedging is it can be wasteful. Multiple calls could be made and all succeed. Only the first result is used and the rest are discarded.
+
+## Configure a gRPC hedging policy
+
+A hedging policy is configured like a retry policy. Note that a hedging policy can't be combined with a retry policy.
+
+```csharp
+var defaultMethodConfig = new MethodConfig
+{
+    Names = { MethodName.Default },
+    HedgingPolicy = new HedgingPolicy
+    {
+        MaxAttempts = 5,
+        NonFatalStatusCodes = { StatusCode.Unavailable }
+    }
+};
+
+var channel = GrpcChannel.ForAddress("https://localhost:5001", new GrpcChannelOptions
+{
+    ServiceConfig = new ServiceConfig { MethodConfigs = { defaultMethodConfig } }
+});
+```
+
+### gRPC hedging options
+
+The following table describes options for configuring gRPC hedging policies:
+
+| Option | Description |
+| ------ | ----------- |
+| MaxAttempts | The hedging policy will send up to this number of calls. Max attempts represents the total number of all attempts, including the original attempt. This value is limited by `GrpcChannelOptions.MaxRetryAttempts` which defaults to 5. Value is required and must be greater than 1. |
+| HedgingDelay | The first call will be sent immediately, but the subsequent hedging calls will be delayed by this value. When the delay is set to zero all hedged calls are sent immediately. |
+| NonFatalStatusCodes | A collection of status codes which indicate other hedge calls may still succeed. If a non-fatal status code is returned by the server, hedged calls will continue. Otherwise, outstanding requests will be canceled and the error returned to the app. For more information about status codes, see [Status codes and their use in gRPC](https://grpc.github.io/grpc/core/md_doc_statuscodes.html). |
```

Comment:
```suggestion
| `NonFatalStatusCodes` | A collection of status codes which indicate other hedge calls may still succeed. If a non-fatal status code is returned by the server, hedged calls will continue. Otherwise, outstanding requests will be canceled and the error returned to the app. For more information about status codes, see [Status codes and their use in gRPC](https://grpc.github.io/grpc/core/md_doc_statuscodes.html). |
```